src/trace-plugin-loader.js

```diff
@@ -0,0 +1,235 @@
+var path = require('path');
+var fs = require('fs');
+var semver = require('semver');
+var PluginAPI = require('./trace-plugin-interface.js');
+
+var plugins = {};
+
+var logger;
+
+/**
+ * Determines the path at which the requested module will be loaded given
+ * the provided parent module.
+ *
+ * @param {string} request The name of the module to be loaded.
+ * @param {object} parent The module into which the requested module will be loaded.
+ */
+function findModulePath(request, parent) {
+  var mainScriptDir = path.dirname(Module._resolveFilename(request, parent));
+  var resolvedModule = Module._resolveLookupPaths(request, parent);
+  var paths = resolvedModule[1];
+  for (var i = 0, PL = paths.length; i < PL; i++) {
+    if (mainScriptDir.indexOf(paths[i]) === 0) {
+      return path.join(paths[i], request.replace('/', path.sep));
+    }
+  }
+  return null;
+}
+
+/**
+ * Determines the version of the module located at `modulePath`.
+ *
+ * @param {?string} modulePath The absolute path to the root directory of the
+ *    module being loaded. This may be null if we are loading an internal module
+ *    such as http.
+ */
+function findModuleVersion(modulePath, load) {
+  if (modulePath) {
+    var pjson = path.join(modulePath, 'package.json');
+    if (fs.existsSync(pjson)) {
+      return load(pjson).version;
+    }
+  }
+  return process.version;
+}
+
+function resolvePluginPath(pluginFile) {
+  // If path doesn't start with '/', './', or '../',
+  // it must refer to a module in node_modules
+  if (!pluginFile.match(/^(\/|\.\/|\.\.\/)/)) {
+    return pluginFile;
+  } else {
+    if (path.isAbsolute(pluginFile)) {
+      return pluginFile;
+    } else {
+      var entryPoint = process.argv[1];
+      return path.join(path.dirname(entryPoint), pluginFile);
+    }
+  }
+}
+
+function checkLoadedModules() {
+  for (var moduleName in plugins) {
+    // \\ is benign on unix and escapes \\ on windows
+    var regex = new RegExp('node_modules\\' + path.sep + moduleName +
+      '\\' + path.sep);
+    for (var file in require.cache) {
+      if (file.match(regex)) {
+        logger.error(moduleName + ' tracing might not work as ' + file +
+            ' was loaded before the trace agent was initialized.');
+        break;
+      }
+    }
+  }
+  if (process._preload_modules && process._preload_modules.length > 0) {
+    var first = process._preload_modules[0];
+    if (first !== '@google/cloud-trace') {
+      logger.error('Tracing might not work as ' + first +
+            ' was loaded with --require before the trace agent was initialized.');
+    }
+  }
+}
+
+function activate(agent) {
+  if (logger) {
+    logger.error('Plugins activated more than once.');
+  }
+  logger = agent.logger;
+
+  // Create a new object exposing functions to create trace spans and propagate
+  // context. This relies on functions currently exposed by the agent.
+  var api = new PluginAPI(agent);
+  var pluginConfig = agent.config().plugins;
```

Comment:
Yes, that is my biggest concern. We have our own framework and my goal is to simply add the creation of the root context when we route initially without having to monkey patch anything. But yes, being able to correlate contexts would be a big win for us, so we are really excited about this. Thanks!